r/R/filesystem.R

```diff
@@ -273,33 +273,53 @@ FileSystem$from_uri <- function(uri) {
   fs___FileSystemFromUri(uri)
 }
 
-get_path_and_filesystem <- function(x, filesystem = NULL) {
+get_paths_and_filesystem <- function(x, filesystem = NULL) {
```

Comment:
What if you left `get_path_and_filesystem()` (singular) as it was and just lapply over the vector of paths where you need multiple?

Reply:
I attempted that first, but that caused multiple errors, prompting me to do it this way

Reply:
Also, there are some other places in the package where `get_path_and_filesystem()` is called where its assertion of singularity is important. I thought about possibly adding an argument like `allow_multiple` but it seemed much cleaner to have singular and plural variants of the function.

Reply:
Ah, I had missed that you retained a singular variant